### PR TITLE
Frontend: Add unit tests for featured challenge & my hosted challenges controller

### PR DESCRIPTION
**Pull request by Sanji515**
**PR contains the front-end unit tests of:**

- Featured Challenge Controller (`featuredChallengeCtrl.js`)
- Hosted Challenge Controller (`hostedChallengeCtrl.js`)

### CODE REVIEW

File: frontend/tests/controllers-test/featuredChallengeCtrl.test.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        it('when challenge logo image is null `challenges/challenge/<challenge_id>/`', function () {
+            challengeSuccess = true;
+            challengePhaseSuccess = null;
+            challengePhaseSplitSuccess = null;
+            successResponse = {
+                is_active: true,
+                image: null
+            };
+            vm = createController();
+            expect(vm.page.image).toEqual("dist/images/logo.png");
```
Comment (galipremsagar) [inline frontend/tests/controllers-test/featuredChallengeCtrl.test.js]: missed validating `successResponse`?

Comment (Sanji515) [reply frontend/tests/controllers-test/featuredChallengeCtrl.test.js]: @galipremsagar done

File: frontend/tests/controllers-test/hostedChallengeCtrl.test.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        it('get the hosted challenge details backend error', function () {
+            hostTeamSuccess = true;
+            hostedChallengeSuccess = false
+            // host team details response
+            successResponse = {
+                results: [
+                    {
+                        id: 1,
+                        team_name: "Host team name",
+                        team_url: "Host team url",
```
Comment (galipremsagar) [inline frontend/tests/controllers-test/hostedChallengeCtrl.test.js]: Put a valid url.

Comment (Sanji515) [reply frontend/tests/controllers-test/hostedChallengeCtrl.test.js]: done

**Comment by galipremsagar** [comment]
@RishabhJain2018  LGTM. Can be merged.

**Pull request merged**